geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala

```diff
@@ -354,9 +354,13 @@ class AccumuloDataStore(val connector: Connector,
 
     GeoMesaTable.getTables(sft).foreach { table =>
       val name = getTableName(sft.getTypeName, table)
-      val deleter = connector.createBatchDeleter(name, auths, numThreads, defaultBWConfig)
-      table.deleteFeaturesForType(sft, deleter)
-      deleter.close()
+      if (table == Z3Table && tableOps.exists(name)) {
```

Comment:
need to move the 'exists' check down to the next line, otherwise we would create a batch deleter against a table that doesn't exist and it would throw an exception.